src/kafkerl_broker_connection.erl

```diff
@@ -344,33 +346,28 @@ handle_fetch_response(Bin, Fetch,
 
 handle_produce_response(Bin, State = #state{connector = Connector, name = Name,
                                             ets = EtsName}) ->
-  case kafkerl_protocol:parse_produce_response(Bin) of
-    {ok, CorrelationId, Topics} ->
-      case ets:lookup(EtsName, CorrelationId) of
-        [{CorrelationId, Messages}] ->
-          ets:delete(EtsName, CorrelationId),
-          {Errors, Successes} = split_errors_and_successes(Topics),
-          % First, send the offsets and messages that were delivered
-          _ = spawn(fun() ->
-                      notify_success(Successes, Messages, Connector)
-                    end),
-          % Then handle the errors
-          case handle_errors(Errors, Messages, Name) of
-            ignore ->
-              {ok, State};
-            {request_metadata, MessagesToResend} ->
-              kafkerl_connector:request_metadata(Connector),
-              ok = resend_messages(MessagesToResend, Connector),
-              {ok, State}
-          end;
-        _ ->
-          _ = lager:warning("~p was unable to get produce response", [Name]),
-          {error, invalid_produce_response}
+  {ok, CorrelationId, Topics} = kafkerl_protocol:parse_produce_response(Bin),
```

Comment:
The result of `parse_produce_response` doesn't reflect that. It never returns an error. If there is a chance of error, it would probably throw an exception or exit.